pkgs/by-name/so/sobjectizer/package.nix

```diff
@@ -0,0 +1,51 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  callPackage,
+  # attention:
+  #   static and shared are both required for tests building(buildTests)
+  withStatic ? true,
+  withShared ? true,
+  buildExamples ? false,
+  buildTests ? true,
+  buildAll ? false,
```

Comment:
```diff
-  withStatic ? true,
-  withShared ? true,
-  buildExamples ? false,
-  buildTests ? true,
-  buildAll ? false,
+  sobjectizer
+ 
+  withStatic ? stdenv.hostPlatform.isStatic,
+  withShared ? !withStatic,
```

The `sobjectizer` reference is not a mistake here btw 😄 look further down for some cool tricks with `passthru.tests`

Reply:
Please see the comment - https://github.com/NixOS/nixpkgs/pull/403862/files#r2072870963